python/semantic_kernel/core_skills/planning/basic_planner.py

```diff
@@ -0,0 +1,203 @@
+
+ShakespeareSkill.shakespeare
+description: Write in Shakespearean style
+args:
+- input: the input to write about
+
+WriterSkill.EmailTo
+description: Write an email to a recipient
+args:
+- input: the input to write about
+- recipient: the recipient's email address.
+
+[GOAL]
+"Tomorrow is Valentine's day. I need to come up with a few date ideas. She likes Shakespeare so write using his style.
+E-mail these ideas to my significant other"
+
+[OUTPUT]
+    {
+        "input": "Valentine's Day Date Ideas",
+        "subtasks": [
+            {"function": "WriterSkill.Brainstorm"},
+            {"function": "ShakespeareSkill.shakespeare"},
+            {"function": "WriterSkill.EmailTo", "args": {"recipient": "significant_other"}},
+            {"function": "translate", "args": {"language": "French"}}
+        ]
+    }
+
+[AVAILABLE FUNCTIONS]
+{{$available_functions}}
+
+[GOAL]
+{{$goal}}
+
+[OUTPUT]
+"""
+
+
+class BasicPlanner:
+    """
+    Description: Basic planner for the Semantic Kernel.
+
+    Usage:
+        kernel.import_skill(BasicPlanner(), "planner");
+
+    Examples:
+
+    {{planner.create_plan}}
+    {{planner.execute_plan}}
+    """
+
+    @sk_function(
+        description="Creates a plan for the given goal based off the functions that are available in the kernel.",
+        name="create_plan",
+    )
+    async def create_plan_async(
+        self,
+        goal: str,
+        kernel: Kernel,
+        prompt: str = PROMPT,
+    ) -> Plan:
+        """
+        Creates a plan for the given goal based off the functions that
+        are available in the kernel.
+        """
+
+        # Create the semantic function for the planner with the given prompt
+        planner = kernel.create_semantic_function(
+            prompt, max_tokens=1000, temperature=0.8
+        )
+
+        # Get a dictionary of skill names to all native and semantic functions
+        native_functions = kernel.skills.get_functions_view()._native_functions
+        semantic_functions = kernel.skills.get_functions_view()._semantic_functions
+        native_functions.update(semantic_functions)
+
+        # Create a flattened list of all functions
+        all_functions = native_functions
+        skill_names = list(all_functions.keys())
+        all_functions_dict = {}
+        for skill_name in skill_names:
+            for f in all_functions[skill_name]:
+                function_name = f.name
+                all_functions_dict[skill_name + "." + function_name] = f.description
+
+        # Create the [AVAILABLE FUNCTIONS] section of the prompt
+        available_functions_string = ""
+        for name in list(all_functions_dict.keys()):
+            available_functions_string += name + "\n"
+            available_functions_string += (
+                "description: " + all_functions_dict[name] + "\n"
+            )
+            available_functions_string += "args:\n"
+            # TODO: FIGURE OUT HOW TO GET ARGS FROM FUNCTIONS
```

Comment:
`FunctionView` I believe is your solution, looks like it is now available in Python too.